src/bokeh/core/property/either.py

```diff
@@ -81,20 +81,24 @@ def __init__(self, type_param0: TypeOrInst[Property[Any]], *type_params: TypeOrI
         super().__init__(type_param0, *type_params, default=default, help=help)
         for tp in self.type_params:
             self.alternatives.extend(tp.alternatives)
+        self.valids = []
 
     def transform(self, value: Any) -> Any:
-        for param in self.type_params:
+        for i, param in enumerate(self.type_params):
             try:
-                return param.transform(value)
+                # if valid was not executed, or all valids failed - transform one by one
+                # else: run transform on the first param that pass validation
+                if len(self.valids)==0 or not any(self.valids) or self.valids[i]: 
+                    return param.transform(value)
             except ValueError:
                 pass
 
```

Comment:
If we make then change in for `self.valid` then maybe like:
```python
if self.valid: 
    return self.valid.transform(value)

for param in self.type_params:
    try:
        return param.transform(value)
    except ValueError:
        pass
```
Question: is `self.valid` is not None, can `self.valid.transform` fail? If so we might `try...except: pass` around the first conditional, to let it fall through to the loop.